File: Python/model/fine.py
class Fine:
    def __init__(self, plate_number, violations):
        self.plate_number = plate_number
        self.violations = violations

    def get_plate_number(self):
        return self.plate_number

    def get_violations(self):
        return self.violations

    def get_total_amount(self):
        total = 0
        for i in range(len(self.violations)):
            v = self.violations[i]
            total += v.get_fee()
        return total

    def __str__(self):
        text = "Traffic for car " + self.plate_number + "\n"
        text += "Total amount: " + str(int(self.get_total_amount())) + " EGP\n"
        text += "Violations:"
        for i in range(len(self.violations)):
            v = self.violations[i]
            text += "\n- " + v.get_description() + " : " + str(int(v.get_fee())) + " EGP"
        return text

File: JAVA/model/Fine.java
package model;
import java.util.Collections;
import java.util.List;

public class Fine {
    private final String plateNumber;
    private final List<Violation> violations;
    public Fine(String plateNumber, List<Violation> violations) {
        this.plateNumber = plateNumber;
        this.violations = Collections.unmodifiableList(violations);
    }
    public String getPlateNumber() {
        return plateNumber;
    }
    public List<Violation> getViolations() {
        return violations;
    }
    public double getTotalAmount() {
        double total = 0;
        for (int i = 0; i < violations.size(); i++) {
            Violation v = violations.get(i);
            total += v.getFee();
        }
        return total;
    }
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Traffic for car ").append(plateNumber).append("\n");
        sb.append("Total amount: ").append((int) getTotalAmount()).append(" EGP\n");
        sb.append("Violations:");
        for (int i = 0; i < violations.size(); i++) {
            Violation v = violations.get(i);
            sb.append("\n- ").append(v.getDescription()).append(" : ").append((int) v.getFee()).append(" EGP");
        }
        return sb.toString();
    }
}